extensions/standard-processors/processors/SegmentContent.cpp

```diff
@@ -58,7 +59,7 @@ void SegmentContent::onTrigger(core::ProcessContext& context, core::ProcessSessi
 
   size_t max_segment_size{};
   const auto segment_size_str = context.getProperty(SegmentSize, original.get());
-  if (!segment_size_str || !core::DataSizeValue::StringToInt(*segment_size_str, max_segment_size)) {
+  if (!segment_size_str || !core::DataSizeValue::StringToInt(*segment_size_str, max_segment_size) || max_segment_size == 0) {
     throw Exception(PROCESSOR_EXCEPTION, fmt::format("Invalid Segment Size {}", segment_size_str));
```

Comment:
this will print `optional(0)` as shown in the unit test below, which is not nice; suggestion:
```
fmt::format("Invalid Segment Size '{}'", segment_size_str.value_or(""))
```

Reply:
good idea, https://github.com/apache/nifi-minifi-cpp/pull/1879/commits/b78de39d16432f40be3cf18a76d9005b79ffdb92